File: backend/src/main/java/pt/ist/socialsoftware/mono2micro/domain/source/SourceFactory.java
package pt.ist.socialsoftware.mono2micro.domain.source;

import static pt.ist.socialsoftware.mono2micro.domain.source.Source.SourceType.*;

public class SourceFactory {
    private static SourceFactory factory = null;

    public static SourceFactory getFactory() {
        if (factory == null)
            factory = new SourceFactory();
        return factory;
    }

    public Source getSource(String sourceType) {
        switch (sourceType) {
            case ACCESSES:
                return new AccessesSource();
            case TRANSLATION:
                return new TranslationSource();
            default:
                throw new RuntimeException("The type \"" + sourceType + "\" is not a valid source type.");
        }
    }
}

File: frontend/src/models/sources/SourceFactory.ts
import Source, {SourceType} from "./Source";
import AccessesSource from "./AccessesSource";
import TranslationSource from "./TranslationSource";

export abstract class SourceFactory {
    static getSource(source: any) : Source {
        switch (source.type) {
            case SourceType.ACCESSES:
                return new AccessesSource(source);
            case SourceType.IDTOENTITIY:
                return new TranslationSource(source);
            default:
                throw new Error('Type ' + source.type + ' unknown.');
        }
    }
}